Jenkinsfile

```diff
@@ -29,6 +30,77 @@ pipeline {
                 sh 'build/run make -j\$(nproc) test-integration'
             }
         }
+
+        stage('Integration Tests SetUp') {
```

Comment:
I will keep working on this, so far Jenkins doesn't like mixing parallel and non-parallel steps, but i am looking at ways to combine the two steps.